### ISSUE TITLE
[UI] does not prevent creation of cluster when chose Azure cloud and do not fill in required fields

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#21616]
master-head (7/18/19)
_This issue does not occur in v2.2.5_

**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Crease Azure RKE cluster via Rancher UI (fresh, not with a template). Chose Azure cloud option.
- DO NOT fill in any of the four required fields. We are allowed to submit form.
- Create Azure Cluster Tempalte. Chose Azure cloud option.
- DO NOT fill in any of the four required fields. We are allowed to submit form.

**Result:**
The UI lets us create the cluster/cluster template even though we have not filled in the mandatory four fields for Azure cloud option. This used to work in v2.2.5 when creating a cluster (UI would block form submission).
UI should also not allow creation of cluster template that has Azure cloud option picked and required fields not filled in.

**Comment by davidnuzik** [comment]
Version: master-head (8/6/19)

Now we block creation of a cluster if the four required Azure cloud fields are not filled in. This is true whether or not a CT is being used. If a CT doesn't have overrides (data already filled in in the CT) then we can still create a cluster. If a CT has some but not all of these four fields as overrides the UI will block accordingly.

Fresh cluster attempt to create w/o required four Azure Cloud fields:
![image](https://user-images.githubusercontent.com/45179589/62560383-c20c3900-b831-11e9-8fc3-b2d919822c84.png)

However, we have an issue. When we have a CT that has an override for aadClientId (and leave this blank in CT) and have all other 3 fields NOT as overrides and filled in. UI still thinks we need to supply the aadClientSecret. 

This is not the case. The aadClientSecret is stored in the CT and cannot be provided by the user leveraging the tempalte:
![image](https://user-images.githubusercontent.com/45179589/62560884-d4d33d80-b832-11e9-964b-900675c38d3e.png)
As you can see user is only not providing aadClientId

Yet UI says we need to enter aadClientSecret too (which is incorrect)
![image](https://user-images.githubusercontent.com/45179589/62560930-ea486780-b832-11e9-8190-f63cde134804.png)

**Comment by davidnuzik** [comment]
@westlywright if you would like this tracked in a separate issue let me know; it just appeared to be the same area.

**Comment by westlywright** [comment]
@davidnuzik no this is fine. will be putting in a fix soonish.

**Comment by davidnuzik** [comment]
Version: master-head (8/7/19)

Refer to this comment above for checking without using a CT. That is working. https://github.com/rancher/rancher/issues/21616#issuecomment-518761616

I now tested this when using a CT (this was what was broken yesterday) and this is now working correctly.

- Fill out CT with Azure cloud and enter all four required fields. Set an override for addClientId and leave the value empty.

Now I can leverage the template and only get prompted to supply the aadClientId if i didn't fill it in.
![image](https://user-images.githubusercontent.com/45179589/62657621-c9a80c80-b91b-11e9-87fc-c39bcdf3fa3e.png)

Last time we tried it asked for the secret too even though that was defined in CT. This works without any issues now.

When I supply my aadClientId everything works fine and I can leverage Azure cloud features.
